File: LeetCode/src/String/M771_宝石与石头.java
package String;

/**
 * @author Hyperspace
 * @date 2023/07/24
 * @file M771_宝石与石头.java
 */
public class M771_宝石与石头 {
    public int numJewelsInStones(String jewels, String stones) {
        int ans = 0;
        for (char c : stones.toCharArray()) {
            if (jewels.indexOf(c) >= 0) {
                ans++;
            }
        }

        return ans;
    }
}

File: GolangSolutions/String/M771_宝石与石头.go
/*
	@author Hyper
	@date 2023/07/24
	@file M771_宝石与石头.go
*/

package String

func numJewelsInStones(jewels string, stones string) (ans int) {
	for _, s := range stones {
		for _, j := range jewels {
			if s == j {
				ans++
			}
		}
	}

	return
}

func numJewelsInStones2(jewels string, stones string) (ans int) {
	mp := make(map[rune]bool)
	for _, j := range jewels {
		mp[j] = true
	}
	for _, s := range stones {
		if mp[s] {
			ans++
		}
	}

	return
}
